ui-react/packages/atlasmap/src/UI/ConditionalExpressionInput.tsx

```diff
@@ -1,30 +1,82 @@
-import React, { FunctionComponent } from "react";
-import { ToolbarGroup, ToolbarItem, Button } from "@patternfly/react-core";
+import React, { FunctionComponent, useState } from "react";
+import {
+  ToolbarGroup,
+  ToolbarItem,
+  Button,
+  Tooltip,
+} from "@patternfly/react-core";
 import { css, StyleSheet } from "@patternfly/react-styles";
 import {
-  // ExpressionContent,
+  ExpressionContent,
   IExpressionContentProps,
 } from "./ExpressionContent";
 
 const styles = StyleSheet.create({
   toolbarItem: { flex: 1 },
 });
 
-export const ConditionalExpressionInput: FunctionComponent<IExpressionContentProps> = () =>
-  // props,
-  {
-    return (
-      <ToolbarGroup className={css(styles.toolbarItem)} role={"form"}>
-        <ToolbarItem>
-          <Button variant={"plain"} tabIndex={-1}>
+export const ConditionalExpressionInput: FunctionComponent<IExpressionContentProps> = ({
+  executeFieldSearch,
+  mappingExpressionAddField,
+  mappingExpressionClearText,
+  isMappingExpressionEmpty,
+  mappingExpressionInit,
+  mappingExpressionInsertText,
+  mappingExpressionObservable,
+  mappingExpressionRemoveField,
+  mappingExpression,
+  onConditionalMappingExpressionEnabled,
+  onToggleExpressionMode,
+  trailerId,
+}) => {
+  const [, setCondExprEnabled] = useState<boolean>();
```

Comment:
Sorry, this usage of `useState` is wrong. It doesn't make sense in React to have a state and then never using it. Is this some way to force the refresh of the child components? If that's the case, then the child components should be refactored to follow React's flow.

Reply:
agree - it wasnt' needed

Reply:
ended up needing this - see updates